testing/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AsciidoctorPluginSmokeTest.groovy

```diff
@@ -129,6 +129,15 @@ class AsciidoctorPluginSmokeTest extends AbstractPluginValidatingSmokeTest {
                     "This API is incompatible with the configuration cache, which will become the only mode supported by Gradle in a future release. " +
                     "Consult the upgrading guide for further information: ${BASE_URL}/userguide/upgrading_version_7.html#task_project"
             )
+
+            runner.expectDeprecationWarningIf(
+                versionNumber.major >= 4,
```

Comment:
❓ Is there a compatible version of Asciidoctor already or in the works? Do we consider the plugin important enough to keep it working?

Reply:
>❓ Is there a compatible version of Asciidoctor already or in the works?

I don't think so. I created an issue to fix the plugin only yesterday:
- https://github.com/asciidoctor/asciidoctor-gradle-plugin/issues/751

>Do we consider the plugin important enough to keep it working?

I don't know how to answer this question. 

We do use it in our own build, meaning it will break as we transition to 9.0 wrapper, where the method is removed.
https://github.com/gradle/gradle/blob/9123d9c3688103a0439c5e22a30028ecb39dc95d/platforms/documentation/docs/build.gradle.kts#L16

Reply:
If we already know we're unable to remove the method in 9.0, I don't think we should say we will in the deprecation message.

Reply:
Don't know that for certain. Depending on our own desire we could work with Asciidoctor to release a version without the issue before we release 9.0.

I don't feel strongly here, and I agree that it does not cost much to keep the method around until 10.0.